signer/metrics.go

```diff
@@ -0,0 +1,206 @@
+package signer
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	// Variables to calculate Prometheus Metrics
+	previousPrecommitHeight         = int64(0)
+	previousPrevoteHeight           = int64(0)
+	previousPrecommitTime           = time.Now()
+	previousPrevoteTime             = time.Now()
+	previousLocalSignStartTime      = time.Now()
+	previousLocalSignFinishTime     = time.Now()
+	previousLocalEphemeralShareTime = time.Now()
```

Comment:
change request:

The time variables must be synchronized, otherwise there are likely data races. 

[go StartMetrics()](https://github.com/strangelove-ventures/horcrux/pull/93/files#diff-9ba01267c02440d5f34d850290066f2b246faf6d49ebc06ac3f1c8c42494531fR265) is reading values while another goroutine is writing to them.

A mutex may be appropriate but will probably litter the codebase with mu `Lock/Unlock`s.

Reply:
Mutex implemented.